### PR TITLE
DTSPO- 14772 - Changes made for Bootstrap test VM

### PR DESCRIPTION
**Pull request by NickAzureDevops**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

changes made for bootstrap test VM repo https://github.com/hmcts/Boostrap-test-vm

### JIRA link (https://tools.hmcts.net/jira/browse/DTSPO-14773)) ###

https://github.com/hmcts/Boostrap-test-vm

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by nchang90** [comment]
Bootstrap test vm repo created https://github.com/hmcts/Boostrap-test-vm

Virtual machine module PR : https://github.com/hmcts/terraform-module-virtual-machine/pull/34

File: locals.tf [hunk 1]
```diff
@@ -3,21 +3,20 @@ locals {
...
 
   # Dynatrace OneAgent
-  dynatrace_settings = var.dynatrace_hostgroup == null && var.dynatrace_server == null ? "{ \"tenantId\" : \"${var.dynatrace_tenant_id}\" , \"token\" : \"${var.dynatrace_token}\" , \"installerArguments\" : \"--set-network-zone=${var.dynatrace_network_zone}\" }" : var.dynatrace_hostgroup != null && var.dynatrace_server == null ? "{ \"tenantId\" : \"${var.dynatrace_tenant_id}\" , \"token\" : \"${var.dynatrace_token}\" , \"hostGroup\" : \"${var.dynatrace_hostgroup}\" , \"installerArguments\" : \"--set-network-zone=${var.dynatrace_network_zone}\"}" : var.dynatrace_hostgroup == null && var.dynatrace_server != null ? "{ \"tenantId\" : \"${var.dynatrace_tenant_id}\" , \"token\" : \"${var.dynatrace_token}\" , \"server\" : \"${var.dynatrace_server}\" , \"installerArguments\" : \"--set-network-zone=${var.dynatrace_network_zone}\" }" : "{ \"tenantId\" : \"${var.dynatrace_tenant_id}\" , \"token\" : \"${var.dynatrace_token}\" , \"hostGroup\" : \"${var.dynatrace_hostgroup}\" , \"server\" : \"${var.dynatrace_server}\" , \"installerArguments\" : \"--set-network-zone=${var.dynatrace_network_zone}\" }"
-
+  dynatrace_env      = var.env
```
Comment (bancey) [inline locals.tf]: Is there much benefit of mapping dynatrace_env directly to `var.env`? Could we just use `var.env` to reduce potential confusion?

File: locals.tf [hunk 2]
```diff
@@ -3,21 +3,20 @@ locals {
...
   template_file = base64encode(format("%s\n%s", templatefile("${path.module}/${local.bootstrap_vm_script}", {
     UF_INSTALL      = tostring(var.install_splunk_uf),
-    UF_USERNAME     = var.splunk_username,
-    UF_PASSWORD     = var.splunk_password,
-    UF_PASS4SYMMKEY = var.splunk_pass4symmkey,
+    UF_USERNAME     = var.splunk_username == null || var.splunk_username == "" ? data.azurerm_key_vault_secret.splunk_username.value : var.splunk_username
```
Comment (thomast1906) [inline locals.tf:19]: Is this not overly complex if you are planning to add in the keyvault references?

Comment (nchang90) [reply locals.tf:19]: we want to do it so if it is not null or empty it will get it from kv

**Review by subtitleuk — APPROVED** [review]

**Pull request merged**